tests/test-util-functions.js

```diff
@@ -0,0 +1,15 @@
+function waitExits (pool, numberOfExitEventsToWait) {
```

Comment:
This is the real trick to avoid setTimeout in most cases @jerome-benoit @Shinigami92

Reply:
I've created an test-utils.js file in another PR, can you merge it first and reuse it here?

Reply:
I think we could take only the readme from your PR, all the other changes are in overlap with those.
Do you prefer a TestUtils class instead of a set of functions? I don't care too much on that.
Let me know please

Reply:
It's way easier to export/import a class once than to do it on a bunch of functions.

Reply:
Yes I agree on that, I can export a class and keep this method on it, please also look at other changes for feedbacks. 
I will do that  tomorrow maybe or if you want feel free to do that but let me know in case you start.

I am sorry for our overlap

Reply:
> I am sorry for our overlap

No pb.